src/typescript.ts

```diff
@@ -1,33 +1,41 @@
 import type { CompilerOptions } from 'typescript'
 import { resolve, dirname } from 'path'
+import { promises as fsp } from 'fs'
 import { Module } from 'module'
-import { exit, fileExists } from './utils'
+import pc from 'picocolors'
+import { exit, fileExists, memoize } from './utils'
+import { DEFAULT_TS_CONFIG } from './constants'
+import { logger } from './logger'
 
 export type TypescriptOptions = {
-  tsConfigPath: string | undefined
+  tsConfigPath?: string
   tsCompilerOptions: CompilerOptions
 }
 
 let hasLoggedTsWarning = false
-function resolveTypescript(cwd: string): typeof import('typescript') {
-  let ts
-  const m = new Module('', undefined)
-  m.paths = (Module as any)._nodeModulePaths(cwd)
-  try {
-    ts = m.require('typescript')
-  } catch (e) {
-    console.error(e)
-    if (!hasLoggedTsWarning) {
-      hasLoggedTsWarning = true
-      exit(
-        'Could not load TypeScript compiler. Try to install `typescript` as dev dependency',
-      )
+const resolveTypescript = memoize<(cwd: string) => typeof import('typescript')>(
```

Comment:
If it can auto infer the type, then we can define it outside and pass to it

```
function resolveTypescript_(cwd: string): typeof import('typescript') {}
const resolveTypescript = memoize(resolveTypescript)
```